packages/hooks/src/useP2PAdvertInfo.ts

```diff
@@ -0,0 +1,25 @@
+import React from 'react';
+import { useFetch } from '@deriv/api';
+import { TSocketRequestQueryOptions } from '@deriv/api/types';
+
+/**
+ * This custom hook returns the advert info for a specific advert by calling 'p2p_advert_info' endpoint
+ */
+const useP2PAdvertInfo = (id: string, options: TSocketRequestQueryOptions<'p2p_advert_info'>) => {
```

Comment:
clarification: Should we move this hook to @deriv/api -> src -> hooks

Reply:
yeah perhaps we should move this out now 👍

Reply:
ok upon discussing with @niloofar-deriv we will keep this in hooks not in api

Reply:
This should be in api package tho 👀 . Since we are fetching and making the possibility of data transformation here as well

Reply:
@amina-deriv we keep these kind of hooks inside `@deriv/hook` package

Reply:
`@deriv/api` contains hooks such as `useFetch` `usePagination` and ...

Reply:
We have so many hooks that are api related in deriv/api. The ones we wanna modify like grouping/filtering/mapping, we keep it in deriv/hooks. I dont know, just following consistency 👀

Reply:
perhaps in this case for useP2PAdvertInfo since we are doing minimal changes to the data we receive, this should be under api as its a pure api call @niloofar-deriv as I recall, this is the intention for the @deriv/api package

Reply:
Let's check this with @faraz-deriv too :thinking:

Reply:
Can leave it here for now, We are still trying out things with the wallet, Later once we have more confident we can move p2p hooks there as well.

Reply:
alright sure thanks man for the clarification 🙏